modules/extensions/src/views/lite/components/debugger/views/Processing.tsx

```diff
@@ -0,0 +1,138 @@
+import { Icon } from '@blueprintjs/core'
+import sdk from 'botpress/sdk'
+import cx from 'classnames'
+import _ from 'lodash'
+import moment from 'moment'
+import React, { FC, Fragment, useState } from 'react'
+
+import ContentSection from '../../../../../../../../src/bp/ui-shared-lite/ContentSection'
+import ToolTip from '../../../../../../../../src/bp/ui-shared-lite/ToolTip'
+import lang from '../../../../lang'
+import style from '../style.scss'
+
+export const Processing: FC<{ processing: { [activity: string]: sdk.IO.ProcessingEntry } }> = props => {
+  const [expanded, setExpanded] = useState({})
+  const { processing } = props
+  let isBeforeMW = true
+
+  const processed = Object.keys(processing)
+    .map(key => {
+      const [type, name, status] = key.split(':')
+      return { type, name, status, completed: moment(processing[key].date), ...processing[key] }
+    })
+    .map((curr, idx, array) => {
+      return { ...curr, execTime: idx === 0 ? 0 : curr.completed.diff(array[idx - 1].completed) }
+    })
+    .filter(x => x.status !== 'skipped')
+    .reduce((acc, item) => {
+      const lastItem = acc.pop()
```

Comment:
You should use. _.get(.slice[-1], 0) instead of pop(). It'll return a ref to lastItem without mutating the original array, thus remove the need of line 31 and lines 33 to 35